azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -268,18 +269,18 @@ private boolean handleNonReadyStatus() {
     if (Status.isStatusFinished(nodeStatus)) {
       quickFinish = true;
     } else if (nodeStatus == Status.DISABLED) {
-      changeStatus(Status.SKIPPED, time);
+      nodeStatus = changeStatus(Status.SKIPPED, time);
       quickFinish = true;
     } else if (this.isKilled()) {
-      changeStatus(Status.KILLED, time);
+      nodeStatus = changeStatus(Status.KILLED, time);
       quickFinish = true;
     }
 
     if (quickFinish) {
       node.setStartTime(time);
-      fireEvent(Event.create(this, Type.JOB_STARTED, null, false));
```

Comment:
I removed `boolean shouldUpdate` from `Event.java` because it wasn't used anywhere anymore.

It seems to be a historical leftover, for example see this file from 2013 where it was originally used:
https://github.com/azkaban/azkaban/blob/0fe79f570a0bb789cbe33c20d464f3816e1b55d6/src/java/azkaban/execapp/FlowRunner.java#L522 :)